pkg/controller/configmapcainjector/configmapcainjector_controller.go

```diff
@@ -0,0 +1,165 @@
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetAnnotations()[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" ||
+				(e.Meta.GetName() == names.TRUST_BUNDLE_CONFIGMAP && e.Meta.GetNamespace() == names.TRUST_BUNDLE_CONFIGMAP_NS)
+		},
+	}
+
+	err = c.Watch(&source.Kind{Type: cfpMap}, &handler.EnqueueRequestForObject{}, pred)
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMapInjector{}
+
+type ReconcileConfigMapInjector struct {
+	client client.Client
+	scheme *runtime.Scheme
+	status *statusmanager.StatusManager
+}
+
+func (r *ReconcileConfigMapInjector) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling update for ca-certs from %s/%s\n", request.Name, request.Namespace)
+
+	combinedCAbundleConfigMap := &corev1.ConfigMap{}
+	combinedCAbundleConfigMapName := types.NamespacedName{
+		Namespace: names.TRUST_BUNDLE_CONFIGMAP_NS,
+		Name:      names.TRUST_BUNDLE_CONFIGMAP,
+	}
+	err := r.client.Get(context.TODO(), combinedCAbundleConfigMapName, combinedCAbundleConfigMap)
+	if err != nil {
+		log.Println(err)
+		return reconcile.Result{}, err
+	}
+	combinedCAbundleData, err := r.validateTrustBundleConfigMap(combinedCAbundleConfigMap)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	//build a list of configMaps
+	configMapsToChange := []corev1.ConfigMap{}
+
+	//The trusted-ca-bundle changed
+	if request.Name == names.TRUST_BUNDLE_CONFIGMAP && request.Namespace == names.TRUST_BUNDLE_CONFIGMAP_NS {
+
+		listOfConfigMaps := &corev1.ConfigMapList{}
+		err = r.client.List(context.TODO(), &client.ListOptions{}, listOfConfigMaps)
+		if err != nil {
+			return reconcile.Result{}, nil
+		}
+		for _, configMap := range listOfConfigMaps.Items {
+			if configMap.ObjectMeta.Annotations[names.TRUST_BUNDLE_CONFIGMAP_ANNOTATION] == "true" {
+				configMapsToChange = append(configMapsToChange, configMap)
+
+			}
+		}
+	} else {
+		//get the requested object
+		requestedCAbundleConfigMap := &corev1.ConfigMap{}
+		requestedCAbundleConfigMapName := types.NamespacedName{
+			Namespace: request.Namespace,
+			Name:      request.Name,
+		}
+		err = r.client.Get(context.TODO(), requestedCAbundleConfigMapName, requestedCAbundleConfigMap)
+		if err != nil {
+			log.Println(err)
+			return reconcile.Result{}, err
+		}
+		configMapsToChange = append(configMapsToChange, *requestedCAbundleConfigMap)
+	}
+
+	for _, configMap := range configMapsToChange {
+		if configMap.Data == nil {
+			configMap.Data = map[string]string{names.TRUST_BUNDLE_CONFIGMAP_KEY: string(combinedCAbundleData)}
+		} else {
+			configMap.Data[names.TRUST_BUNDLE_CONFIGMAP_KEY] = string(combinedCAbundleData)
+		}
+		err = r.client.Update(context.TODO(), &configMap)
```

Comment:
Because you're touching so many configmaps, this might result in a lot of conflicts. I'd recommend changing this to a patch, as well as using the retryOnConflict helpers.

Reply:
@squeed take a look at my comment here: https://github.com/openshift/cluster-network-operator/pull/274/files#r311696339 and let me know your thoughts on the approach I suggested.

Reply:
This is also not a full kubeClient patch is not implemented. I did put in the RetryOnConflict

Reply:
@JacobTanenbaum @squeed so Patch was apparently added upstream in https://github.com/kubernetes-sigs/controller-runtime/pull/235 which merged 2019-03-19. And server-side patch was added in https://github.com/kubernetes-sigs/controller-runtime/pull/435.

And those are only in controller-runtime 0.2.0-alpha and later. CNO currently vendors operator-sdk 0.4.1 which doesn't have that. And even upstream operator-sdk only locks controller-runtime 0.1.10 https://github.com/operator-framework/operator-sdk/blob/master/go.mod

So until upstream operator-sdk updates, and then CNO revendors from that, we can't get Patch.

Reply:
Bummer. Thanks for looking in to that.